theano/gof/op.py

```diff
@@ -1062,12 +1062,35 @@ def load_c_code(self):
             with open(func_file, 'r') as f:
                 self.func_codes.append(f.read())
 
+        # If both the old section markers and the new section markers are
+        # present, raise an error because we don't know which ones to follow.
+        old_markers_present = False
+        new_markers_present = False
+        for code in self.func_codes:
+            if self.backward_re.search(code):
+                old_markers_present = True
+            if self.section_re.search(code):
+                new_markers_present = True
+
+        if old_markers_present and new_markers_present:
+            raise ValueError('Both the new and the old syntax for '
+                             'identifying code sections are present in the '
+                             'provided C code. These two syntaxes should not '
+                             'be used at the same times')
+
         self.code_sections = dict()
         for i, code in enumerate(self.func_codes):
-            if ('THEANO_APPLY_CODE_SECTION' in code or
-                'THEANO_SUPPORT_CODE_SECTION' in code):
+            if self.backward_re.search(code):
                 # This is backward compat code that will go away in a while
+
+                # Check for code outside of the supported sections
                 split = self.backward_re.split(code)
+                if split[0].strip() != '':
```

Comment:
Was this check done in the old code? Or this is just a port from the new code?

Other then this questions, the PR seem good to me.